libbeat/outputs/elasticsearch/client.go

```diff
@@ -417,17 +417,18 @@ func (client *Client) bulkCollectPublishFails(result eslegclient.BulkResult, dat
 				encodedEvent := data[i].EncodedEvent.(*encodedEvent)
 				if encodedEvent.deadLetter {
 					stats.nonIndexable++
-					client.log.Errorf("Can't deliver to dead letter index event (status=%v). Enable debug logs to view the event and cause.", status)
-					client.log.Debugf("Can't deliver to dead letter index event %#v (status=%v): %s", data[i], status, msg)
+					client.log.Errorf("Can't deliver to dead letter index event (status=%v). Look at the event log to view the event and cause.", status)
+					client.log.Errorw(fmt.Sprintf("Can't deliver to dead letter index event %#v (status=%v): %s", data[i], status, msg), "log.type", "event")
 					// poison pill - this will clog the pipeline if the underlying failure is non transient.
 				} else if client.deadLetterIndex != "" {
-					client.log.Warnf("Cannot index event (status=%v), trying dead letter index. Enable debug logs to view the event and cause.", status)
-					client.log.Debugf("Cannot index event %#v (status=%v): %s, trying dead letter index", data[i], status, msg)
+					client.log.Warnf("Cannot index event (status=%v), trying dead letter index. Look at the event log to view the event and cause.", status)
+					client.log.Warnw(fmt.Sprintf("Cannot index event %#v (status=%v): %s, trying dead letter index", data[i], status, msg), "log.type", "event")
 					client.setDeadLetter(encodedEvent, status, string(msg))
+
 				} else { // drop
 					stats.nonIndexable++
-					client.log.Warnf("Cannot index event (status=%v): dropping event! Enable debug logs to view the event and cause.", status)
-					client.log.Debugf("Cannot index event %#v (status=%v): %s, dropping event!", data[i], status, msg)
+					client.log.Warnf("Cannot index event (status=%v): dropping event! Look at the event log to view the event and cause.", status)
+					client.log.Warnw(fmt.Sprintf("Cannot index event %#v (status=%v): %s, dropping event!", data[i], status, msg), "log.type", "event")
```

Comment:
Can `log.type` be a constant with some documentation about what it's purpose is?

Right now somebody reading this log line won't know the significance of this without viewing the PR that introduced it.

Perhaps some developer documentation on logging is a good idea as well, but just documenting the meaning of `log.type` well would solve most of this problem.

Reply:
> Can log.type be a constant with some documentation about what it's purpose is?

Reply:
Done: https://github.com/elastic/elastic-agent-libs/pull/202